aom/src/main/java/com/nedap/archie/rminfo/MetaModels.java

```diff
@@ -121,12 +121,11 @@ public void selectModel(String rmPublisher, String rmPackage, String rmRelease)
             selectedBmmModel = validationResult == null ? null : validationResult.getModel();
         }
 
-        for(AomProfile profile:aomProfiles.getProfiles()) {
-            if(profile.getProfileName().equalsIgnoreCase(rmPublisher)) {
-                this.selectedAomProfile = profile;
-                break;
-            }
+        AomProfile profile = getAomProfileWithSchemaId(selectedBmmModel);
```

Comment:
I think directly setting `this.selectedAomProfile`, without the local variable `profile` is clearer.

Reply:
Yes, but then we need to do a ```this.selectedAomProfile = null;``` as a first line. I could instead do:

```
this.selectedAomProfile = getAomProfile()
```

Then add a new method that does the current logic :)

Reply:
I was thinking of:
```java
        this.selectedAomProfile = getAomProfileWithSchemaId(selectedBmmModel);
        if(this.selectedAomProfile == null) {
            this.selectedAomProfile = getAomProfileOnPublisher(rmPublisher);
        }
```
I think this should work.

Reply:
Oh of course. done :)